File: solution/0000-0099/0001.Two Sum/Solution.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var m = [Int: Int]()
        var i = 0
        while true {
            let x = nums[i]
            let y = target - nums[i]
            if let j = m[target - nums[i]] {
                return [j, i]
            }
            m[nums[i]] = i
            i += 1
        }
    }
}

File: solution/0000-0099/0001.Two Sum/Solution.php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $m = [];
        foreach ($nums as $i => $x) {
            $y = $target - $x;
            if (isset($m[$y])) {
                return [$m[$y], $i];
            }
            $m[$x] = $i;
        }
    }
}